### PR TITLE
ToMask Transform

### PR DESCRIPTION
**Pull request by olliethomas**

### CODE REVIEW

File: src/data.py [hunk 1]
```diff
@@ -195,7 +216,7 @@ def __init__(
         self.crops = torch.as_tensor(data["crop"])
 
         # THe transformation applied to the mask
-        self._target_transform = ToTensor()
+        self._target_transform = transforms.Compose([ToMask(), ToTensor()])
```
Comment (MylesBartlett) [inline src/data.py:219]:
```suggestion
        self._target_transform = Compose([ToMask(), ToTensor()])
```

File: src/data.py [hunk 2]
```diff
@@ -122,6 +122,27 @@ def _patches_from_image_mask_pair(
...
+class ToMask:
+    def __init__(self):
+        self.mapping = {
+            (0, 0, 0): 0,  # Target 0 (background)
+            (216, 124, 18): 0,  # Target 0 (background)
+            (255, 255, 255): 1,  # Target 1 (crop)
+            (216, 67, 81): 2,  # Target 2 (weed)
+        }
+
+    def __call__(self, mask_img: Image) -> np.ndarray:
```
Comment (MylesBartlett) [inline src/data.py:134]:
```suggestion
    def __call__(self, mask_img: Image.Image) -> np.ndarray:
```

File: main.py [hunk 3]
```diff
@@ -29,6 +28,7 @@ def experiment(
     epochs: int = typer.Option(100, "--epochs"),
     use_amp: bool = typer.Option(False, "--use-amp"),
     seed: Optional[int] = typer.Option(47, "--seed"),
+    download: bool = typer.Option(False, "--dl"),
```
Comment (MylesBartlett) [inline main.py:31]:
```suggestion
    download: bool = typer.Option(False, "--download", "-dl"),
```

File: src/data.py [hunk 4]
```diff
@@ -306,7 +330,7 @@ def __getitem__(self, index: int) -> Union[TrainBatch, TestBatch]:
         crop = self.crops[index]
         if self.mask_fps is not None:
             mask_t = Image.open(self._dataset_folder / self.mask_fps[index])
-            mask = self._target_transform(mask_t)
+            mask = self._target_transform(mask_t).squeeze(0)
```
Comment (olliethomas) [inline src/data.py:333]:
```suggestion
            mask = self._target_transform(mask_t)
```
I'm not sure this if this is needed....

Comment (olliethomas) [reply src/data.py:333]: Well, it runs without it, so 🤷‍♂️

**Pull request merged**